File: livingrimoire start here/LivinGrimoire java/src/Auxiliary_Modules/AXSkillBundle.java
package Auxiliary_Modules;

import LivinGrimoire.*;

import java.util.ArrayList;

public class AXSkillBundle {
    private final ArrayList<Skill> skills = new ArrayList<>();
    private final Neuron tempN = new Neuron();
    private Kokoro kokoro = new Kokoro(new AbsDictionaryDB());

    public void setKokoro(Kokoro kokoro) {
        this.kokoro = kokoro;
        for (Skill skill : skills) {
            skill.setKokoro(this.kokoro);
        }
    }

    public AXSkillBundle(Skill...skillsParams) {
        for (Skill skill : skillsParams)
        {
            skill.setKokoro(this.kokoro);
            skills.add(skill);
        }
    }
    public AXSkillBundle addSkill(Skill skill){
        // builder pattern
        skill.setKokoro(this.kokoro);
        skills.add(skill);
        return this;
    }
    public AlgorithmV2 dispenseAlgorithm(String ear, String skin, String eye){
        for (Skill skill : skills) {
            skill.input(ear, skin, eye);
            skill.output(tempN);
            for (int j = 1; j < 6; j++) {
                Algorithm temp = tempN.getAlg(j);
                if (temp != null) {
                    return new AlgorithmV2(j,temp);
                }
            }
        }
        return null;
    }
    public int getSize(){return skills.size();}
}

File: livingrimoire start here/LivinGrimoire Kotlin/kotlin/auxiliary_modules/AXSkillBundle.kt
package auxiliary_modules

import livinGrimoire.*

class AXSkillBundle(vararg skillsParams: Skill) {
    private val skills: ArrayList<Skill> = ArrayList<Skill>()
    private val tempN: Neuron = Neuron()
    private var kokoro: Kokoro = Kokoro(AbsDictionaryDB())
    fun setKokoro(kokoro: Kokoro) {
        this.kokoro = kokoro
        for (skill in skills) {
            skill.kokoro = this.kokoro
        }
    }

    init {
        for (skill in skillsParams) {
            skill.kokoro = this.kokoro
            skills.add(skill)
        }
    }

    fun addSkill(skill: Skill): AXSkillBundle {
        // builder pattern
        skill.kokoro = this.kokoro
        skills.add(skill)
        return this
    }

    fun dispenseAlgorithm(ear: String, skin: String, eye: String): AlgorithmV2? {
        for (skill in skills) {
            skill.input(ear, skin, eye)
            skill.output(tempN)
            for (j in 1..5) {
                val temp: Algorithm? = tempN.getAlg(j)
                if (temp != null) {
                    return AlgorithmV2(j, temp)
                }
            }
        }
        return null
    }

    val size: Int
        get() = skills.size
}